src/openzaak/components/documenten/api/serializers.py

```diff
@@ -616,9 +624,20 @@ def update(self, instance, validated_data):
             )
         else:
             # The serialiser .create() method does not support nested data, so these have to be added separately
-            instance.integriteit = integriteit
-            instance.ondertekening = ondertekening
-            instance.save()
+            update_fields = []
+            if integriteit is not None:
+                instance.integriteit = integriteit
+                update_fields.append("integriteit_algoritme")
+                update_fields.append("integriteit_waarde")
+                update_fields.append("integriteit_datum")
+
+            if ondertekening is not None:
+                instance.ondertekening = ondertekening
+                update_fields.append("ondertekening_soort")
+                update_fields.append("ondertekening_datum")
```

Comment:
Let's avoid multiple append calls here 
```suggestion
                update_fields.extend(["integriteit_algoritme", "integriteit_waarde", "integriteit_datum"])
                
            if ondertekening is not None:
                instance.ondertekening = ondertekening
                update_fields.extend(["ondertekening_soort", "ondertekening_datum"])
```